### PR TITLE
Refactor Brush

### PR DESCRIPTION
**Pull request by billyjanitsch**
This is part of my work towards #84.

Removes `withControlledState` from `Brush` and simplifies the implementation.

### CODE REVIEW

**Comment by nabeel-** [comment]
Looking over the code I can't pinpoint exactly where the bug is but the brush will shrink if you leave the bounds of the page and then come back into it. I also verified this is not the behavior on master.

brush

**Comment by billyjanitsch** [comment]
Rats. I noticed some weird behavior with this chart too but I thought I was able to reproduce it back to 1.4.4. I'll play around with it some more -- thanks for catching.

File: src/extensions/Brush.js [hunk 1]
```diff
@@ -17,201 +15,124 @@ const BRUSH_ELEMENT_NAMES = [
...
-  if (props.brushElementName === 'brushesCenter') {
-    const bounds = domainToBounds(props, childProps)
-    const mBounds = {
-      x1: bounds.x1 - childProps.mouseDelta.x,
-      x2: bounds.x2 - childProps.mouseDelta.x,
-      y1: bounds.y1 - childProps.mouseDelta.y,
-      y2: bounds.y2 - childProps.mouseDelta.y,
-    }
-    if (isOutOfBounds(mBounds, childProps.rootProps.plotRect)) {
-      const constrainedBounds = constraintToPlotRect(bounds, childProps)
```
Comment (billyjanitsch) [inline src/extensions/Brush.js]: Note to self: this is called with `bounds`, not `mBounds` (if the rect is dragged outside the chart, we shouldn't squish it down)

**Comment by billyjanitsch** [comment]
Thanks, that fix lgtm. @nabeel- want to approve and merge?

**Pull request merged**